content/examples/02-next-pages/tools/render-interface-during-tool-call.mdx

```diff
@@ -0,0 +1,238 @@
+            return 'result' in toolInvocation ? (
+              toolInvocation.toolName === 'getWeatherInformation' ? (
+                <div
+                  key={toolCallId}
+                  className="flex flex-col gap-2 p-4 bg-blue-400 rounded-lg"
+                >
+                  <div className="flex flex-row justify-between items-center">
+                    <div className="text-4xl text-blue-50 font-medium">
+                      {toolInvocation.result.value}°
+                      {toolInvocation.result.unit === 'celsius' ? 'C' : 'F'}
+                    </div>
+
+                    <div className="h-9 w-9 bg-amber-400 rounded-full flex-shrink-0" />
+                  </div>
+                  <div className="flex flex-row gap-2 text-blue-50 justify-between">
+                    {toolInvocation.result.weeklyForecast.map(
+                      (forecast: any) => (
+                        <div
+                          key={forecast.day}
+                          className="flex flex-col items-center"
+                        >
+                          <div className="text-xs">{forecast.day}</div>
+                          <div>{forecast.value}°</div>
+                        </div>
+                      ),
+                    )}
+                  </div>
+                </div>
+              ) : toolInvocation.toolName === 'getLocation' ? (
+                <div
+                  key={toolCallId}
+                  className="text-gray-500 bg-gray-100 rounded-lg p-4"
+                >
+                  User is in {toolInvocation.result}.
+                </div>
+              ) : (
+                <div key={toolCallId} className="text-gray-500">
+                  Tool call {`${toolInvocation.toolName}: `}
+                  {toolInvocation.result}
+                </div>
+              )
+            ) : (
+              <div key={toolCallId} className="text-gray-500">
+                Calling {toolInvocation.toolName}...
+              </div>
+            );
+          })}
+        </div>
+      ))}
+
+      <form onSubmit={handleSubmit}>
+        <input
+          className="fixed bottom-0 w-full max-w-md p-2 mb-8 border border-gray-300 rounded shadow-xl"
+          value={input}
+          placeholder="Say something..."
+          onChange={handleInputChange}
+        />
+      </form>
+    </div>
+  );
+}
+```
+
+## Server
+
+```tsx filename='api/chat.ts'
+import { openai } from '@ai-sdk/openai';
+import { convertToCoreMessages, streamText } from 'ai';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { z } from 'zod';
+
+export default async function handler(
+  request: NextApiRequest,
+  response: NextApiResponse,
+) {
+  const { messages } = await request.body;
+
+  const result = await streamText({
+    model: openai('gpt-4-turbo'),
+    messages: convertToCoreMessages(messages),
+    tools: {
+      // server-side tool with execute function:
+      getWeatherInformation: {
```

Comment:
can we wrap this in `tool` (and import `tool`) and remove the types in line 204?